src/signTx.c

```diff
@@ -789,7 +830,72 @@ static void signTx_handleCertificate_ui_runStep()
 
 		advanceCertificatesStateIfAppropriate();
 	}
-	UI_STEP_END(HANDLE_INPUT_STEP_INVALID);
+	UI_STEP_END(HANDLE_CERTIFICATE_STEP_INVALID);
+}
+
+#ifdef POOL_OPERATOR_APP
+enum {
+	HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_DISPLAY_OPERATION = 650,
+	HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_DISPLAY_EPOCH,
+	HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_CONFIRM,
+	HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_RESPOND,
+	HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_INVALID,
+};
+
+static void signTx_handleCertificatePoolRetirement_ui_runStep()
+{
+	TRACE("UI step %d", ctx->ui_step);
+	ASSERT(ctx->stageData.certificate.type == CERTIFICATE_TYPE_STAKE_POOL_RETIREMENT);
+	ui_callback_fn_t* this_fn = signTx_handleCertificatePoolRetirement_ui_runStep;
+
+	UI_STEP_BEGIN(ctx->ui_step);
+
+	UI_STEP(HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_DISPLAY_OPERATION) {
+		char poolId[200];
+		explicit_bzero(poolId, SIZEOF(poolId));
+
+		size_t length = encode_hex(
+		                        ctx->stageData.certificate.poolKeyHash, SIZEOF(ctx->stageData.certificate.poolKeyHash),
+		                        poolId, SIZEOF(poolId)
+		                );
+		ASSERT(length == strlen(poolId));
+		ASSERT(length == 2 * SIZEOF(ctx->stageData.certificate.poolKeyHash));
+
+		ui_displayPaginatedText(
+		        "Retire stake pool",
+		        poolId,
+		        this_fn
+		);
+	}
+	UI_STEP(HANDLE_CERTIFICATE_POOL_RETIREMENT_STEP_DISPLAY_EPOCH) {
+		ui_displayUint64Screen(
+		        "Epoch",
```

Comment:
I'd print a more user-friendly title clarifying when  should that retirement actually take effect, something like "Since end of epoch" - based on:
https://docs.cardano.org/projects/cardano-node/en/latest/stake-pool-operations/retire_stakepool.html